testing/api/test_api.py

```diff
@@ -224,25 +225,150 @@ def local_get_devices():
   r = requests.get(f"{API}/system/interfaces", timeout=5)
+  # Parse the JSON response
   response = r.json()
+  # Retrieve the actual network interfaces
   local_interfaces = get_network_interfaces()
-  assert set(response.keys()) == set(local_interfaces)
 
-  # schema expects a flat list
+  # Check if status code is 200 (OK)
+  assert r.status_code == 200, f"status code is {r.status_code}"
+  # Check if the key are in the response
+  assert set(response.keys()) == set(local_interfaces)
+  # Ensure that all values in the response are strings
   assert all(isinstance(x, str) for x in response)
 
+def test_update_system_config(testrun, restore_config): # pylint: disable=W0613
+  """Test update system configuration endpoint ('/system/config')"""
 
-def test_status_idle(testrun): # pylint: disable=W0613
-  until_true(
-      lambda: query_system_status().lower() == "idle",
-      "system status is `idle`",
-      30,
+  # Configuration data to update
+  updated_system_config = {
+      "network": {
+          "device_intf": "updated_endev0a",
+          "internet_intf": "updated_wlan1"
+      },
+      "log_level": "DEBUG"
+  }
+
+  # Send the post request to update the system configuration
+  r = requests.post(f"{API}/system/config",
+                    data=json.dumps(updated_system_config),
+                    timeout=5)
+
+  # Check if status code is 200 (OK)
+  assert r.status_code == 200
+
+  # Parse the JSON response
+  response = r.json()
+
+  # Check if the response["network"]["device_intf"] has been updated
+  assert (
+    response["network"]["device_intf"]
+    == updated_system_config["network"]["device_intf"]
+  )
+
+  # Check if the response["network"]["internet_intf"] has been updated
+  assert (
+        response["network"]["internet_intf"]
+        == updated_system_config["network"]["internet_intf"]
+  )
+
+  # Check if the response["log_level"] has been updated
+  assert (
+    response["log_level"]
+    == updated_system_config["log_level"]
+  )
+
+def test_update_system_config_invalid_config(testrun, restore_config): # pylint: disable=W0613
+  """Test invalid configuration file for update system configuration"""
+
+  # Configuration data to update
+  updated_system_config = {
+      "network": {
+          "device_intf": "updated_endev0a",
+          "internet_intf": "updated_wlan1"
+      }
+  }
+
+  # Send the post request to update the system configuration
+  r = requests.post(f"{API}/system/config",
+                    data=json.dumps(updated_system_config),
+                    timeout=5)
+
+  # Check if status code is 400 (Invalid config)
+  assert r.status_code == 400
+
+def test_get_system_config(testrun): # pylint: disable=W0613
+  """Tests get system configuration endpoint ('/system/config')"""
+  # Send a GET request to the API to retrieve system configuration
+  r = requests.get(f"{API}/system/config", timeout=5)
+
+  # Open the local system configuration file
+  with open(
```

Comment:
If there is a method to open JSON then use it, if not create one. I think there is one called load_json?